scripts/respeaker_node.py

```diff
@@ -2,438 +2,9 @@
-        if self.timer_led and self.timer_led.is_alive():
-            self.timer_led.shutdown()
-        self.timer_led = rospy.Timer(rospy.Duration(3.0),
-                                       lambda e: self.respeaker.set_led_trace(),
-                                       oneshot=True)
-
-    def on_audio(self, data, channel):
-        self.pub_audios[channel].publish(AudioData(data=data))
-        if channel == self.main_channel:
-            self.pub_audio.publish(AudioData(data=data))
-            if self.is_speeching:
-                if len(self.speech_audio_buffer) == 0:
-                    self.speech_audio_buffer = self.speech_prefetch_buffer
-                self.speech_audio_buffer += data
-            else:
-                self.speech_prefetch_buffer += data
-                self.speech_prefetch_buffer = self.speech_prefetch_buffer[-self.speech_prefetch_bytes:]
-
-    def on_timer(self, event):
-        stamp = event.current_real or rospy.Time.now()
-        is_voice = self.respeaker.is_voice()
-        doa_rad = math.radians(self.respeaker.direction - 180.0)
-        doa_rad = angles.shortest_angular_distance(
-            doa_rad, math.radians(self.doa_yaw_offset))
-        doa = math.degrees(doa_rad)
-
-        # vad
-        if is_voice != self.prev_is_voice:
-            self.pub_vad.publish(Bool(data=is_voice))
-            self.prev_is_voice = is_voice
-
-        # doa
-        if doa != self.prev_doa:
-            self.pub_doa_raw.publish(Int32(data=doa))
-            self.prev_doa = doa
-
-            msg = PoseStamped()
-            msg.header.frame_id = self.sensor_frame_id
-            msg.header.stamp = stamp
-            ori = T.quaternion_from_euler(math.radians(doa), 0, 0)
-            msg.pose.position.x = self.doa_xy_offset * np.cos(doa_rad)
-            msg.pose.position.y = self.doa_xy_offset * np.sin(doa_rad)
-            msg.pose.orientation.w = ori[0]
-            msg.pose.orientation.x = ori[1]
-            msg.pose.orientation.y = ori[2]
-            msg.pose.orientation.z = ori[3]
-            self.pub_doa.publish(msg)
-
-        # speech audio
-        if is_voice:
-            self.speech_stopped = stamp
-        if stamp - self.speech_stopped < rospy.Duration(self.speech_continuation):
-            self.is_speeching = True
-        elif self.is_speeching:
-            buf = self.speech_audio_buffer
-            self.speech_audio_buffer = str()
-            self.is_speeching = False
-            duration = 8.0 * len(buf) * self.respeaker_audio.bitwidth
-            duration = duration / self.respeaker_audio.rate / self.respeaker_audio.bitdepth
-            rospy.loginfo("Speech detected for %.3f seconds" % duration)
-            if self.speech_min_duration <= duration < self.speech_max_duration:
-
-                self.pub_speech_audio.publish(AudioData(data=buf))
 
+import rospy
+from respeaker_ros import RespeakerNode
 
 if __name__ == '__main__':
     rospy.init_node("respeaker_node")
```

Comment:
@furushchev Most of the code in this file was moved into `node.py` inside the module (to let people import and customize the node in their own packages). Because the diff was large GitHub collapsed it making it look like the file was removed completely, but it's still here, just very small now.